src/main/java/edu/usf/cutr/grha/io/CsvParser.java

```diff
@@ -23,20 +27,14 @@
 
 public abstract class CsvParser {
 
-    /**
-     * Creates a reader for a resource in the relative path.
-     * Fetched from https://github.com/uniVocity/univocity-parsers/blob/master/src/test/java/com/univocity/parsers/examples/Example.java#L39
-     *
-     * @param filePath relative path of the resource to be read
-     *
-     * @return a reader of the resource, or null if the file couldn't be read
-     */
-    public Reader getReader(String filePath) {
-      ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-      InputStream inputStream = classLoader.getResourceAsStream(filePath);
-      if (inputStream != null) {
-        return new InputStreamReader(inputStream, StandardCharsets.UTF_8);
-      }
-      return null;
+    public <T> List<T> parseFile(BeanListProcessor<T> beanListProcessor, InputStream inputStream) {
```

Comment:
Good job finding common implementation to share here! It makes sense to keep CsvParser then.